lib/ecl/ecl_sum_data.c

```diff
@@ -1205,25 +1203,40 @@ void ecl_sum_data_fwrite_interp_csv_line(const ecl_sum_data_type * data, time_t
   ecl_sum_data_init_interp_from_sim_time(data, sim_time, &time_index1, &time_index2, &weight1, &weight2);
 
   for (int i = 0; i < num_keywords; i++) {
-    double value = ecl_sum_data_vector_iget( data, sim_time, keylist , i , time_index1, time_index2, weight1, weight2);
+    int params_index = ecl_sum_vector_iget_param_index(keylist, i);
+    if (params_index >= 0) {
+      bool is_rate = ecl_sum_vector_iget_is_rate(keylist, i);
+      double value = ecl_sum_data_vector_iget( data, sim_time, params_index , is_rate, time_index1, time_index2, weight1, weight2);
 
-    if (i == 0)
-      fprintf(fp, "%f", value);
-    else
-      fprintf(fp, ",%f", value);
+      if (i == 0)
+        fprintf(fp, "%f", value);
+      else
+        fprintf(fp, ",%f", value);
+    } else {
+      if (i == 0)
+        fputs("", fp);
+      else
+        fputs(",", fp);
+    }
   }
 }
 
 void ecl_sum_data_get_interp_vector( const ecl_sum_data_type * data , time_t sim_time, const ecl_sum_vector_type * keylist, double_vector_type * results){
   int num_keywords = ecl_sum_vector_get_size(keylist);
   double weight1, weight2;
   int    time_index1, time_index2;
+  double missing_value = -1;
 
   ecl_sum_data_init_interp_from_sim_time(data, sim_time, &time_index1, &time_index2, &weight1, &weight2);
   double_vector_reset( results );
   for (int i = 0; i < num_keywords; i++) {
-    double value = ecl_sum_data_vector_iget( data, sim_time, keylist , i , time_index1, time_index2, weight1, weight2);
-    double_vector_iset( results, i , value );
+    int params_index = ecl_sum_vector_iget_param_index(keylist, i);
+    if (params_index >= 0) {
+      bool is_rate = ecl_sum_vector_iget_is_rate(keylist, i);
+      double value = ecl_sum_data_vector_iget( data, sim_time, params_index, is_rate, time_index1, time_index2, weight1, weight2);
+      double_vector_iset( results, i , value );
+    } else
+      double_vector_iset( results, i , missing_value);
```

Comment:
I'm not sure I understand this part (I'm sure I don't understand this part) but is it safe to set the values to -1?  Is it necessary?

Reply:
> I'm not sure I understand this part (I'm sure I don't understand this part) but is it safe to set the values to -1? Is it necessary?

Good question - I was slightly uncomfortable doing that. The point of the whole PR is that if the keyword vector asks for keywords: `["KEY1", "KEY2", "KEY3"]` and the summary instance in question is missing `KEY2` we still need to return a result with three elements - to ensure that things are correctly aligned. As matters stand it is actyally the `_fprintf( )` function which is the most important - in that case the printed result should:

```txt
100,,200
```

i.e. an empty element in the csv output. We could:

1. [Maybe] just remove the offending function.
2. Pass in a user supplied value to be used in the case of missing values.
3. Use a different value than `-1` - e.g. `nan`.
4. Return an additional mask-like structure to indicate which values are valid.